File: Tests/Pages/HistoryPage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;
using Tests.Data;

namespace Tests.Pages
{
    public class HistoryPage : BasePage
    {
        public override string PageUrl => "History";
        public override string PageName => "History";

        private IWebElement CalculatorBtn => WebDriver.Driver.FindElement(By.XPath("//div[text() = 'Calculator']"));
        private IWebElement ClearBtn => WebDriver.Driver.FindElement(By.Id("clear"));
        private IWebElement Table => WebDriver.Driver.FindElement(By.Id("history"));

        public List<List<string>> History => Table
            .FindElements(By.XPath("//tr[td]"))
            .Select(row => row
                .FindElements(By.TagName("TD"))
                .Select(cell => cell.Text).ToList())
            .ToList();

        public override void Open()
        {
            base.Open();

            try
            {
                new WebDriverWait(WebDriver.Driver, TimeSpan.FromSeconds(Defaults.ImplicitWait))
                    .Until(driver => driver.FindElements(By.XPath("//table//tr")).Any());
            }
            catch { /**/ }

        }

        public void Clear() => ClearBtn.Click();
        public void ReturnToCalculator() => CalculatorBtn.Click();
    }
}

File: Tests/pages/HistoryPage.py
from time import sleep
from selenium.webdriver.common.by import By

from pages.BasePage import BasePage


class HistoryPage(BasePage):
    def __init__(self, driver):
        super().__init__(driver)
        sleep(1)

    def is_history_empty(self): return self.get_history_size() == 0
    def get_history_size(self): return len(self.find_elements(By.XPATH, "//table//tr[td]"))
    def get_history(self): return [self.get_history_row(row) for row in range(1, self.get_history_size()+1)]

    def get_history_row(self, index):
        row = self.find_element(By.XPATH, "//table//tr[td][" + str(index) + "]")
        cells = row.find_elements(By.XPATH, ".//td")
        return [cell.text for cell in cells]

    def clear(self):
        from pages.DepositPage import DepositPage
        self.find_element(By.ID, "clear").click()
        return DepositPage(self.driver)
